### ISSUE TITLE
Deprecate `get_field`

### ISSUE THREAD

**Issue by jarednova** [timber/timber#1827]
- [x] Deprecate `Core::get_field()` (in favor of meta)
- [x] Deprecate any other class's `::get_field()` (in favor of meta)
- [x] Ensure docs reflect `meta` > `get_field`
- [x] Add to upgrade guide
- [ ] ~Implement defensive [`__get()` solution](https://github.com/timber/timber/pull/1618#issuecomment-356441760) proposed by @gchtr~

**Comment by palmiak** [comment]
But what about ACF integration? Beacuse most user, when using ACF, would probably like to use `post.get_field` - maybe we should add the alias only when ACF is installed. What do you think?

**Comment by palmiak** [comment]
@gchtr ok - I fully understand and I think this is a great approach - especially the separation of integrations.

But let's go a bit further with ACF example. `get_field` has 3 parameters - `id`, `field_name` and `format_value` - in case i would like to set `format_value` to false how should I approach this? 
What do you think about adding the second parameter to `meta()` called `$args` with this, it would be easier for everybody to add his own parameters for integrations `post.meta( 'some_field', { 'format': false } )`.

**Comment by palmiak** [comment]
@gchtr I did some quick research on the plugins that are similar to ACF (creating metaboxes and displaying them). 

**Plugins that use only one parameter:**
- Custom Field Suite
- ultimate-fields

**Plugins that use more than one parameter:**
- ACF - but as you saw we can use the `raw_meta` - although IMO it would be more logical to use the same function and different arguments
- Toolset is using `echo(types_render_field( 'field-slug', array( 'arg1' => 'val1', 'arg2' => 'val2' ) ));`
- Pods - `pods_field ( $pod, $id = false, $name = null, $single = false )`
- Carbon Fields - `carbon_get_post_meta( $id, $name, $container_id = '' )`
- Metabox.io - `rwmb_meta( $field_id, $args, $post_id )`
- Papi - `papi_get_field( $meta, $slug, $default )`

**Plugins using native `get_post_meta`:**
- CMB2
- http://fieldmanager.org/
- Piklist
- Custom Post Types and Custom Fields creator – WCK

In short - two most popular plugins are using more than one argument - so I think it would be safer and easier for integration creators if we could provider them with the possibility to pass all the parameters. 

On the other hand your argument about what would happen if two plugins would use the same parameter is valid. But it would only existed when two integrations would be hooked to `meta_filter`. But in this case we have a bigger problem than parameters - the trouble would be with two plugins trying to do the same thing.

**Comment by gchtr** [comment]
Thanks for digging into how the different plugins handle getting meta values! 😍

> In short - two most popular plugins are using more than one argument - so I think it would be safer and easier for integration creators if we could provide them with the possibility to pass all the parameters.

Cool! I’m really starting to like this approach.

> On the other hand your argument about what would happen if two plugins would use the same parameter is valid. But it would only exist when two integrations would be hooked to `meta_filter`. But in this case we have a bigger problem than parameters - the trouble would be with two plugins trying to do the same thing.

Exactly! We can probably forget about two plugins doing the same thing. There might be other plugins that use meta data and that could hook into this function. WooCommerce, maybe? The primary use case is still plugins that only handle custom field. So I think we should consider those for now.

I’ll starting working on a pull request.

**Comment by gchtr** [comment]
Resolved by #1861.

All acceptance criteria for this issue are resolved, except for the following one:

- Implement defensive [`__get()` solution](https://github.com/timber/timber/pull/1618#issuecomment-356441760)

With the new recommended approach to load custom field values via `meta()`, this would probably be an edge case anyways. If there are many issues that relate to this after we release 2.0, we can still react with better debugging output.
